modules/promotion/src/Plugin/Commerce/PromotionOffer/OrderPercentageOff.php

```diff
@@ -18,9 +18,11 @@ class OrderPercentageOff extends PercentageOffBase {
    */
   public function execute() {
     $order = $this->getOrder();
-    $adjustment_amount = $order->getTotalPrice()->multiply($this->getAmount());
-    $adjustment_amount = $this->rounder->round($adjustment_amount);
-    $this->applyAdjustment($order, $adjustment_amount);
+    if ($order->getTotalPrice()) {
```

Comment:
This sounds like a bug where the order should _always_ return a total price. I say we remove this line, prove test fails and fix order entity for a proper TDD approach.